src/System.Private.CoreLib/shared/System/Type.Helpers.cs

```diff
@@ -94,7 +95,7 @@ public bool IsVisible
                         return false;
 
                     // this should be null for non-nested types.
-                    type = type.DeclaringType;
+                    type = type.DeclaringType!;
```

Comment:
@jaredpar here we had to use the `!` operator as well because, `DeclaringType` is nullable, however the condition on this while checks for `IsNested` which returns `DeclaringType != null` -- so in this line, `type.DeclaringType` will never be null. This is another case where the compiler is not flowing enough data to know this will not be null.

Reply:
im wondering if this is even realistically possible for compiler to fix except for some very simple cases